File: app/src/main/java/vertretunggut/app/niklas/vertretungsplan/RepPlanFrame.java
package vertretunggut.app.niklas.vertretungsplan;

import android.support.design.widget.FloatingActionButton;
import android.view.View;

/**
 * Created by nwuensche on 26.09.16.
 */
public class RepPlanFrame {
    private MainActivity activity;
    private FloatingActionButton prevDay;
    private FloatingActionButton nextDay;
    private static boolean nextDayButtonLastPressed = false;

    public RepPlanFrame(MainActivity activity) {
        this.activity = activity;
        prevDay = (FloatingActionButton) activity.findViewById(R.id.prev_day_button);
        nextDay = (FloatingActionButton) activity.findViewById(R.id.next_day_button);

        prevDay.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                previousDayButtonPressed();
            }
        });

        nextDay.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                nextDayButtonPressed();
            }
        });
    }

    public void enableMoveButtons() {
        nextDay.show();
        prevDay.show();
    }

    public void setTitle(String headerTitle){
        activity.getSupportActionBar().setTitle(headerTitle);
    }

    public void disableLastPressedButton() {
        if (nextDayButtonLastPressed()) {
            nextDay.hide();
        } else {
            prevDay.hide();
        }
    }

    public boolean nextDayButtonLastPressed() {
        return nextDayButtonLastPressed;
    }

    public void previousDayButtonPressed() {
        activity.decreaseCurrentRepPlanSite();
        nextDayButtonLastPressed = false;

        boolean connected = handleNetworkRight();

        if(connected) {

            activity.restartRepPlanGetter();
        }
    }

    public void nextDayButtonPressed() {

        activity.increaseCurrentRepPlanSite();
        nextDayButtonLastPressed = true;

        boolean connected = handleNetworkRight();

        if(connected){
            activity.restartRepPlanGetter();
        }

    }

    public boolean handleNetworkRight(){
        if(!NoNetworkHandler.isNetworkAvailable(activity)) {
            new NoNetworkHandler(activity).showNoNetworkView();
            return false;
        }
        return true;
    }
}

File: app/src/main/java/vertretunggut/app/niklas/vertretungsplan/RepPlanFrame.kt
package vertretunggut.app.niklas.vertretungsplan

import android.support.design.widget.FloatingActionButton
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by nwuensche on 26.09.16.
 */
class RepPlanFrame(private val activity: MainActivity) {
    private val prevDay = activity.prev_day_button
    private val nextDay = activity.next_day_button

    init {
        prevDay.setOnClickListener { previousDayButtonPressed() }
        nextDay.setOnClickListener { nextDayButtonPressed() }
    }

    fun enableMoveButtons() {
        nextDay.show()
        prevDay.show()
    }

    fun setTitle(headerTitle: String) {
        activity.supportActionBar!!.title = headerTitle
    }

    fun disableLastPressedButton() {
        if (nextDayButtonLastPressed()) {
            nextDay.hide()
        } else {
            prevDay.hide()
        }
    }

    fun nextDayButtonLastPressed(): Boolean {
        return nextDayButtonLastPressed
    }

    fun previousDayButtonPressed() {
        activity.decreaseCurrentRepPlanSite()
        nextDayButtonLastPressed = false

        val connected = handleNetworkRight()

        if (connected) {
            activity.restartRepPlanGetter()
        }
    }

    fun nextDayButtonPressed() {

        activity.increaseCurrentRepPlanSite()
        nextDayButtonLastPressed = true

        val connected = handleNetworkRight()

        if (connected) {
            activity.restartRepPlanGetter()
        }
    }

    fun handleNetworkRight(): Boolean {
        if (!NoNetworkHandler.isNetworkAvailable(activity)) {
            NoNetworkHandler(activity).showNoNetworkView()
            return false
        }
        return true
    }

    companion object {
        private var nextDayButtonLastPressed = false
    }
}
